app/controllers/books_controller.rb

```diff
@@ -45,11 +45,18 @@ def by_category
     @category = ::Category.find_by(name: params[:name])
   end
 
-  def by_category
-    @category = ::Category.find_by(name: params[:name])
-  end
 
   private
+  def adult?
```

Comment:
można lepiej:

```
def adult?
  ((Date.today - current_user.birghtday) / 365).to_i > 19
end
```

Pytanie, czy jest Ci potrzebne wywołanie:
```
@user = current_user
```

Sprawdź, czy zadziała bez tego. Jeśli koniecznie potrzebujesz mieć tę zmienną `@user`, to warto tę jedną linijkę wydzielić do before_action.

Reply:
Ale i tak ta metoda powinna być w klasie `User`, o czym będzie jeszcze niżej.